File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/PoliticaReservaBO.java

package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.politicas.model.PoliticaReserva;
import java.rmi.Remote;
import java.rmi.RemoteException;

public interface PoliticaReservaBO extends Remote{
    
    public Integer insertar(Integer cantMaxHorasDeRecojo,Integer cantMaxReservasAgendadasPorUsuario) throws RemoteException;

    public Integer modificar(Integer idPoliticaReserva,Integer cantMaxHorasDeRecojo,Integer cantMaxReservasAgendadasPorUsuario) throws RemoteException;

    public Integer eliminar(Integer idPoliticaReserva) throws RemoteException;

    public ArrayList<PoliticaReserva> listarTodos(Integer limite) throws RemoteException;

    public PoliticaReserva obtenerPorId(Integer idPoliticaReserva) throws RemoteException;
    
}

File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPoliticaBO/PoliticaReservaBO.cs
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftBibliotecaPoliticaReservaBO
{
    public class PoliticaReservaBO : BaseBO
    {
        public int insertar(int cantMaxHorasDeRecojo, int cantMaxReservasAgendadasPorUsuario)
        {
            return this.WsCliente.politicareserva_insertar(cantMaxHorasDeRecojo, cantMaxReservasAgendadasPorUsuario);
        }

        public int modificar(int idPoliticaReserva, int cantMaxHorasDeRecojo, int cantMaxReservasAgendadasPorUsuario)
        {
            return this.WsCliente.politicareserva_modificar(idPoliticaReserva, cantMaxHorasDeRecojo, cantMaxReservasAgendadasPorUsuario);
        }

        public int eliminar(int idPoliticaReserva)
        {
            return this.WsCliente.politicareserva_eliminar(idPoliticaReserva);
        }

        public BindingList<politicaReserva> listarTodos(int limite)
        {
            try
            {
                politicaReserva[] politicasReserva = this.WsCliente.politicareserva_listarTodos(limite);
                if (politicasReserva == null) return new BindingList<politicaReserva>();

                return new BindingList<politicaReserva>(politicasReserva);
            }
            catch (Exception e)
            {
                return new BindingList<politicaReserva>();
            }
        }

        public politicaReserva obtenerPorId(int idPoliticaReserva)
        {
            return this.WsCliente.politicareserva_obtenerPorId(idPoliticaReserva);
        }
    }
}
